src/Stryker.Core/Stryker.Core/Initialisation/InitialisationProcess.cs

```diff
@@ -79,26 +82,59 @@ public MutationTestInput Initialize(StrykerOptions options)
                     options.MsBuildPath);
             }
 
-            InitializeDashboardProjectInformation(options, projectInfo);
+            InitializeDashboardProjectInformation(options, _projectInfo);
 
             if (_testRunner == null)
             {
-                _testRunner = new VsTestRunnerPool(options, projectInfo);
+                _testRunner = new VsTestRunnerPool(options, _projectInfo);
             }
 
             var input = new MutationTestInput
             {
-                ProjectInfo = projectInfo,
-                AssemblyReferences = _assemblyReferenceResolver.LoadProjectReferences(projectInfo.ProjectUnderTestAnalyzerResult.References).ToList(),
+                ProjectInfo = _projectInfo,
+                AssemblyReferences = _assemblyReferenceResolver.LoadProjectReferences(_projectInfo.ProjectUnderTestAnalyzerResult.References).ToList(),
                 TestRunner = _testRunner,
             };
 
             return input;
         }
 
-        public InitialTestRun InitialTest(StrykerOptions options) =>
+        public InitialTestRun InitialTest(StrykerOptions options)
+        {
             // initial test
-            _initialTestProcess.InitialTest(options, _testRunner);
+            var result = _initialTestProcess.InitialTest(options, _testRunner);
+
+            if (_testRunner.DiscoverTests().Count != 0)
+            {
+                return result;
+            }
+            // no test have been discovered, diagnose this
+            DiagnoseLackOfDetectedTest(_projectInfo);
+            throw new InputException("No test has been detected. Make sure your test project contains test and is compatible with VsTest.");
+        }
+
+        private static readonly Dictionary<string, string> TestFrameworks = new()
+        {
+            ["xunit.core"] = "xunit.runner.visualstudio",
+            ["nunit.framework"] = "NUnit3.TestAdapter",
+            ["Microsoft.VisualStudio.TestPlatform.TestFramework"] = "Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter"
+        };
+
+        public static void DiagnoseLackOfDetectedTest(ProjectInfo projectInfo)
+        {
+            foreach (var testProject in projectInfo.TestProjectAnalyzerResults)
+            {
+                foreach (var (framework, adapter) in TestFrameworks)
+                {
+                    if (testProject.References.Any(r => r.Contains(framework)) && !testProject.References.Any(r => r.Contains(adapter)))
+                    {
+                        throw new InputException($"Project '{testProject.ProjectFilePath}' is not supported by VsTest because it is missing an appropriate VstTest adapter for '{framework}'. " +
```

Comment:
While I agree with this I am surprised you went with this option since you were so vehemently against it. I expected you to go with a warning instead of an error as discussed. Nevertheless, I don't expect this to be a big problem.

Reply:
This is exactly what I intended all along: as of now, Stryker stops if no test is detected and provides hints for known/supported test libraries.
I was against doing this preventively, i.e. before trying to detect some tests.
I sensed we had trouble understanding each other, hence I provided this PR as a better basis for conversation 😄 .

Reply:
Ah I see, fair point :)